### PR TITLE
Add docs for Mac workflow setup

### PR DESCRIPTION
**Pull request by bduffany**
Sending out for early review -- to be submitted after the following:

- [x] https://github.com/buildbuddy-io/buildbuddy-internal/pull/978 is in prod
- [x] https://github.com/buildbuddy-io/buildbuddy/pull/1237 is is released
- [x] https://github.com/buildbuddy-io/buildbuddy/pull/1242 is released
- Test workflows using `test //...` from `rules_swift` repo
  - [x] amd64
  - [x] arm64 (M1)
---

**Version bump**: Patch

### CODE REVIEW

File: docs/workflows-config.md [hunk 1]
```diff
@@ -58,6 +58,57 @@ Other points to note:
   problems caused by using conflicting versions of Bazel in different
   build environments.
 
+## Mac configuration
+
+By default, workflows will execute on BuildBuddy's shared Linux executors,
+but it is also possible to run workflows on MacOS by using self-hosted
```
Comment (brentleyjones) [inline docs/workflows-config.md:64]:
```suggestion
but it is also possible to run workflows on macOS by using self-hosted
```

File: docs/workflows-config.md [hunk 2]
```diff
@@ -58,6 +58,57 @@ Other points to note:
...
+        ...
+```
+
+2. If you haven't already, [enable workflows for your
+   repo](/docs/workflows-setup#enable-workflows-for-a-repo), then create a
+   file called `buildbuddy.yaml` at the root of your repo. See the
+   [Example config](#example-config) for a starting point.
+
+3. Set `os: "darwin"` on the workflow action that you would like to build
+   on MacOS. Note: if you copy another action as a starting point, be sure
```
Comment (brentleyjones) [inline docs/workflows-config.md:90]:
```suggestion
   on macOS. Note: if you copy another action as a starting point, be sure
```

File: docs/workflows-config.md [hunk 3]
```diff
@@ -81,6 +132,10 @@ A named group of Bazel commands that run when triggered.
 - **`name`** (`string`): A name unique to this config, which shows up as the name of the check
   in GitHub.
 - **`triggers`** ([`Triggers`](#triggers)): The triggers that should cause this action to be run.
+- **`os`** (`string`): The operating system on which to run the workflow.
+  Defaults to `"linux"`. `"darwin"` (MacOS) is also supported, but
```
Comment (brentleyjones) [inline docs/workflows-config.md:136]:
```suggestion
  Defaults to `"linux"`. `"darwin"` (macOS) is also supported, but
```

File: docs/workflows-config.md (hunk 1 above)
Comment (bduffany) [reply docs/workflows-config.md]: Thanks! Done

File: docs/workflows-config.md [hunk 4]
```diff
@@ -58,6 +58,57 @@ Other points to note:
...
+        </dict>
+        ...
+```
+
+2. If you haven't already, [enable workflows for your
+   repo](/docs/workflows-setup#enable-workflows-for-a-repo), then create a
+   file called `buildbuddy.yaml` at the root of your repo. See the
+   [Example config](#example-config) for a starting point.
+
+3. Set `os: "darwin"` on the workflow action that you would like to build
```
Comment (tylerwilliams) [inline docs/workflows-config.md:89]: nit: add an optional #4 to explain setting arch for m1 or old style?

Comment (bduffany) [reply docs/workflows-config.md:89]: Done (kept it with step 3 to keep it short)

File: docs/workflows-config.md [hunk 5]
```diff
@@ -81,6 +134,10 @@ A named group of Bazel commands that run when triggered.
 - **`name`** (`string`): A name unique to this config, which shows up as the name of the check
   in GitHub.
 - **`triggers`** ([`Triggers`](#triggers)): The triggers that should cause this action to be run.
+- **`os`** (`string`): The operating system on which to run the workflow.
```
Comment (tylerwilliams) [inline docs/workflows-config.md:137]: if you add arch here too then i think we'll be good

Comment (bduffany) [reply docs/workflows-config.md:137]: Done

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Comment by bduffany** [comment]
FYI, I updated the docs to mention the minimum required RAM for workflows.

**Pull request merged**